src/components/PostsList.tsx

```diff
@@ -2,85 +1,104 @@
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import { PostDetails } from './PostDetails';
+import classNames from 'classnames';
+import { getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  posts: Post[];
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({ posts }) => {
+  const [open, setOpen] = useState(true);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(false);
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+  const handleOpenPost = (post: Post) => {
+    if (selectedPost?.id === post.id) {
+      setOpen(true);
+      setSelectedPost(null);
+    } else {
+      setOpen(false);
+      setSelectedPost(post);
+      setLoading(true);
 
-          <td data-cy="PostTitle">
-            fugit voluptas sed molestias voluptatem provident
-          </td>
+      getPostComments(post.id)
+        .then(fetchedComments => setComments(fetchedComments))
+        .catch(() => {
+          setError('Something went wrong');
```

Comment:
Consider displaying the error message to the user when an error occurs during comment fetching. This will improve the user experience by informing them of what went wrong.